### PR TITLE
Feature/restructure stylesheets

### PR DESCRIPTION
**Pull request by pzi**
## Restructure all stylesheets

Adopted new structure as it was done in other, newish projects recently. Will update https://github.com/thefrontiergroup/standards later on to reflect the same structure.

Also did some minor fixes. But the whole thing is in dire need of an overhaul. In due time...

### CODE REVIEW

**Comment by jordanmaguire** [comment]
>  But the whole thing is in dire need of an overhaul

Can you expand on that a teeny bit - what's wrong with it as is?

File: Gemfile [hunk 1]
```diff
@@ -12,20 +12,21 @@ gem 'devise'
...
+gem 'neat'
 gem 'normalize-rails'
+gem 'page_title_helper'
 gem 'sass-rails'
 gem 'simple_form'
 gem 'uglifier'
-gem 'bourbon'
-gem 'neat'
-gem 'page_title_helper'
-gem 'active_link_to'
```
Comment (jordanmaguire) [inline Gemfile]: alphabeticool 👍

File: app/assets/stylesheets/abstracts/_functions.sass [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/// @return {color}
+@function white($opacity)
+  @return rgba(255, 255, 255, $opacity)
+
+/// Helper to mix given colour with white
+/// @param {color} $color
+/// @param {Number} $amount
+/// @return {color}
+@function whiten($color, $amount: 50%)
+  @if unitless($amount) == true or unit($amount) != "%"
```
Comment (jordanmaguire) [inline app/assets/stylesheets/abstracts/_functions.sass]: Can this  just be  `@if unitless($amount) or unit($amount) != "%"`

File: app/assets/stylesheets/abstracts/_functions.sass [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @error "$amount expects a percentage. Value was: #{$amount}."
+  @return mix(#fff, $color, $amount)
+
+/// Helper to mix given colour with (brand) black
+/// @param {color} $color
+/// @param {Number} $amount
+/// @param {color} $color [$brand-black]
+/// @return {color}
+@function blacken($color, $amount: 0.5, $base-black-color: $brand-black)
+  @if unitless($amount) == true or unit($amount) != "%"
```
Comment (jordanmaguire) [inline app/assets/stylesheets/abstracts/_functions.sass]: same hurrrrr

**Comment by jordanmaguire** [comment]
Looks good :+1:

File: app/assets/stylesheets/abstracts/_functions.sass (hunk 2 above)
Comment (pzi) [inline app/assets/stylesheets/abstracts/_functions.sass]: Good call.

**Comment by pzi** [comment]
>Can you expand on that a teeny bit - what's wrong with it as is?

The code is not modular/componentized enough. Too dependant on certain classes/html elements in the source. A number of selectors are too long and too restrictive. Heading hierarchy is better but the all the typography styles need to be improved. These kind of things.

**Pull request merged**